docs/tasks/administer-cluster/kubeadm-upgrade-ha.md

```diff
@@ -0,0 +1,179 @@
+[Creating HA clusters with kubeadm](/docs/setup/independent/high-availability/) for instructions on how to do this. The upgrade procedure described here targets clusters created following those very instructions. You may also want to read [Upgrading/downgrading kubeadm clusters between v1.8 to v1.9](/docs/tasks/administer-cluster/kubeadm-upgrade-1-9/) for more information on upgrading clusters with `kubeadm`.
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+Before proceeding:
+
+- You need to have a functional `kubeadm` HA cluster running version 1.8.0 or higher in order to use the process described here. Swap also needs to be disabled.
+- Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md) carefully.
+- Note that `kubeadm upgrade` will not touch any of your workloads, only Kubernetes-internal components. As a best-practice you should back up what's important to you. For example, any app-level state, such as a database an app might depend on (like MySQL or MongoDB) must be backed up beforehand.
+
+**Caution:** All the containers will get restarted after the upgrade, due to container spec hash value gets changed.
+{: .caution}
+
+Also, note that only one minor version upgrade is supported. For example, you can only upgrade from 1.8 to 1.9, not from 1.7 to 1.9.
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Preparation
+
+Before actually starting the upgrade, start with some preparation. For the actual upgrade `kubeadm` of at least the target version is required:
+
+```shell
+# Use the latest stable release or manually specify a
+# released Kubernetes version
+$ export VERSION=$(curl -sSL https://dl.k8s.io/release/stable.txt) 
+$ export ARCH=amd64 # or: arm, arm64, ppc64le, s390x
+$ curl -sSL https://dl.k8s.io/release/${VERSION}/bin/linux/${ARCH}/kubeadm > /tmp/kubeadm
+$ chmod a+rx /tmp/kubeadm
+```
+
+Copy this file to _/tmp_ on your primary master if necessary. Now run this command for checking prerequisites and determining the versions you'd get:
+
+```shell
+$ /tmp/kubeadm upgrade plan
+```
+
+If the prerequisites are met you'll get a summary of the software versions kubeadm would upgrade to, like this:
+
+    Upgrade to the latest stable version:
+
+    COMPONENT            CURRENT   AVAILABLE
+    API Server           v1.8.3    v1.9.2
+    Controller Manager   v1.8.3    v1.9.2
+    Scheduler            v1.8.3    v1.9.2
+    Kube Proxy           v1.8.3    v1.9.2
+    Kube DNS             1.14.5    1.14.7
+    Etcd                 3.2.7     3.1.11
+
+Note that upgrading etcd is not supported here because in the supported configuration it is running externally, hence if necessary you'll have to upgrade it according to etcd's upgrade instruction which is beyond scope here.
```

Comment:
This may warrant a {: caution} callout - https://kubernetes.io/docs/home/contribute/style-guide/#caution

recommended wording update:
- "which is beyond scope here" -> "which is beyond the scope of these instructions."